platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorEditText.kt

```diff
@@ -5,32 +5,34 @@ import android.content.ClipboardManager
 import android.content.Context
 import android.content.Context.CLIPBOARD_SERVICE
 import android.os.Build
+import android.os.Parcelable
+import android.text.Selection
 import android.text.Spannable
 import android.text.SpannableStringBuilder
 import android.util.AttributeSet
 import android.view.KeyEvent
 import android.view.MotionEvent
 import android.view.inputmethod.EditorInfo
 import android.view.inputmethod.InputConnection
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.findViewTreeViewModelStoreOwner
 import com.google.android.material.textfield.TextInputEditText
+import io.element.android.wysiwyg.inputhandlers.InterceptInputConnection
 import io.element.android.wysiwyg.inputhandlers.models.EditorInputAction
 import io.element.android.wysiwyg.inputhandlers.models.InlineFormat
-import io.element.android.wysiwyg.inputhandlers.InputProcessor
-import io.element.android.wysiwyg.inputhandlers.InterceptInputConnection
 import io.element.android.wysiwyg.utils.EditorIndexMapper
+import io.element.android.wysiwyg.viewmodel.EditorViewModel
 import uniffi.wysiwyg_composer.MenuState
-import uniffi.wysiwyg_composer.newComposerModel
 
 class EditorEditText : TextInputEditText {
 
     private var inputConnection: InterceptInputConnection? = null
-    private val inputProcessor = InputProcessor(
-        context,
-        menuStateCallback = { menuStateChangedListener?.menuStateChanged(it) },
-        // Using the returned ComposerModel automatically loads the native libraries and will crash
-        // layout preview and other tools. We're making it nullable as a workaround for that.
-        composer = if (isInEditMode) null else newComposerModel()
-    )
+    private val viewModel: EditorViewModel by lazy {
+        val viewModelStoreOwner = findViewTreeViewModelStoreOwner()
```

Comment:
I had this working before by getting the `ViewModelStoreOwner` of the `context` and using a custom key to get the right ViewModel instance. This worked as soon as the EditText was instantiated.

The new approach with `findViewTreeViewModelStoreOwner()` is simpler, but it will only work once the view has been attached to the current window.